include/multipass/utils.h

```diff
@@ -97,6 +98,18 @@ void validate_server_address(const std::string& value);
 bool valid_hostname(const std::string& name_string);
 std::string generate_mac_address();
 bool valid_mac_address(const std::string& mac);
+template <typename NetworkContainer>
```

Comment:
What is the point of making this template parameter here where all the callers pass `std::vector<NetworkInterfaceInfo>` type? Is there an envision for `std::list` or `std::deque`?

Reply:
Hmm, I am not sure to be honest. Notice this is essentially moving existing code. I suppose it was just a generic implementation that ended up being used with vectors only. Do you think we should change it?

Reply:
OK, I have changed this.